### ISSUE TITLE
[Android] Update Lottie After Upgrade >= RN0.60

### ISSUE THREAD

**Issue by jdnichollsc** [expo/expo#5559]
### Environment

```
jdnichollsc% expo diagnostics

  Expo CLI 3.0.10 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.15.3 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.11.2 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5692245
      Xcode: 10.3/10G8 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^34.0.4 => 34.0.4 
      react: 16.8.6 => 16.8.6 
      react-native: https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.1 => 3.11.1 
    npmGlobalPackages:
      expo-cli: 3.0.10
```

**Target:** Android

### Steps to Reproduce
Load Lottie animations after install lottie with the documentation of Expo https://docs.expo.io/versions/latest/sdk/lottie/

### Expected Behavior
Don't crash the app and execute the animations from Android too

### Actual Behavior
Only works from iOS, Android app is closed with this error:
```
2019-09-06 14:06:25.764 5507-5507/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: host.exp.exponent, PID: 5507
    java.lang.IllegalStateException: Missing values for keyframe.
        at com.a.a.a.b.c.b(FloatKeyframeAnimation.java:16)
        at com.a.a.a.b.c.a(FloatKeyframeAnimation.java:8)
        at com.a.a.a.b.a.e(BaseKeyframeAnimation.java:125)
        at com.a.a.a.b.o.d(TransformKeyframeAnimation.java:113)
        at com.a.a.c.c.a.a(BaseLayer.java:196)
        at com.a.a.c.c.b.b(CompositionLayer.java:100)
        at com.a.a.c.c.a.a(BaseLayer.java:198)
        at com.a.a.g.draw(LottieDrawable.java:312)
        at android.widget.ImageView.onDraw(ImageView.java:1360)
        at android.view.View.draw(View.java:21860)
        at android.view.View.buildDrawingCacheImpl(View.java:21129)
        at android.view.View.buildDrawingCache(View.java:20989)
        at android.view.View.draw(View.java:21578)
        at android.view.ViewGroup.drawChild(ViewGroup.java:4551)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4326)
        at abi34_0_0.com.facebook.react.views.view.ReactViewGroup.dispatchDraw(ReactViewGroup.java:684)
        at android.view.View.updateDisplayListIfDirty(View.java:20719)
        at android.view.View.draw(View.java:21586)
        at android.view.ViewGroup.drawChild(ViewGroup.java:4551)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4326)
        at abi34_0_0.com.facebook.react.views.view.ReactViewGroup.dispatchDraw(ReactViewGroup.java:684)
        at android.view.View.updateDisplayListIfDirty(View.java:20719)
        at android.view.View.draw(View.java:21586)
        at android.view.ViewGroup.drawChild(ViewGroup.java:4551)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4326)
        at android.view.View.draw(View.java:21863)
        at android.widget.ScrollView.draw(ScrollView.java:2768)
        at abi34_0_0.com.facebook.react.views.scroll.ReactScrollView.draw(ReactScrollView.java:400)
        at android.view.View.updateDisplayListIfDirty(View.java:20733)
        at android.view.View.draw(View.java:21586)
        at android.view.ViewGroup.drawChild(ViewGroup.java:4551)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4326)
        at abi34_0_0.com.facebook.react.views.view.ReactViewGroup.dispatchDraw(ReactViewGroup.java:684)
        at android.view.View.draw(View.java:21863)
        at android.view.View.updateDisplayListIfDirty(View.java:20733)
        at android.view.View.draw(View.java:21586)
        at android.view.ViewGroup.drawChild(ViewGroup.java:4551)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4326)
        at abi34_0_0.com.facebook.react.views.view.ReactViewGroup.dispatchDraw(ReactViewGroup.java:684)
        at android.view.View.updateDisplayListIfDirty(View.java:20719)
        at android.view.View.draw(View.java:21586)
        at android.view.ViewGroup.drawChild(ViewGroup.java:4551)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4326)
        at abi34_0_0.com.facebook.react.views.view.ReactViewGroup.dispatchDraw(ReactViewGroup.java:684)
        at android.view.View.updateDisplayListIfDirty(View.java:20719)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4535)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4507)
        at android.view.View.updateDisplayListIfDirty(View.java:20688)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4535)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4507)
        at android.view.View.updateDisplayListIfDirty(View.java:20688)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4535)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4507)
        at android.view.View.updateDisplayListIfDirty(View.java:20688)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4535)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4507)
        at android.view.View.updateDisplayListIfDirty(View.java:20688)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4535)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4507)
        at android.view.View.updateDisplayListIfDirty(View.java:20688)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4535)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4507)
```

### Reproducible Demo
Expo cli install a previous version of Lottie (2.6.1) instead of 3.1.0, check more details of this error here https://github.com/react-native-community/lottie-react-native/issues/473

Thanks in advance for your help! 👍

**Comment by cruzach** [comment]
Fixing 2.6.1 is an upstream `lottie-react-native` issue

Upgrading the to `v3`, however, requires >=RN0.60.0 (current SDK is pinned @ 0.59.8)

**Comment by jdnichollsc** [comment]
Can we downgrade or something like that in the meantime?

**Comment by changLiuUNSW** [comment]
@jdnichollsc I believe your Lottie file is generated with latest lottie 3. As @cruzach said, currently there is no way to fix this.
The workaround now is to re-export the lottie file from bodymovin with old JSON format by changing:
Settings > Advanced > Export Old Json Format
airbnb/lottie-web#1561

**Comment by jdnichollsc** [comment]
Ohh I downloaded the files from https://lottiefiles.com :/

**Comment by jdnichollsc** [comment]
Guys, do you have a deadline to update RN to 0.60? 🙏

**Comment by zeeroj** [comment]
mismo problema cuando solucionaran?

**Comment by cruzach** [comment]
@jdnichollsc Likely SDK 36. See [here](https://expo.canny.io/feature-requests/p/update-to-react-native-060x) for a discussion on that

**Comment by jdnichollsc** [comment]
@cruzach is there any steps to update the SDK?

**Comment by cruzach** [comment]
the [SDK 35 release notes](https://blog.expo.io/expo-sdk-35-is-now-available-beee0dfafbf4) have steps to upgrade to that SDK, but it is still based on RN 0.59. You'll have to wait for SDK 36 for the next RN bump

**Comment by jdnichollsc** [comment]
thanks, waiting for that version 💯  Where can we see the different SDK versions of Expo?

**Comment by cruzach** [comment]
Our [blog](https://blog.expo.io/) has release notes for each one (among other announcements), other than that take a look at our [website](https://expo.io) and [docs](https://docs.expo.io/versions/latest/)

**Comment by SimenB** [comment]
Was this updated for 36? I can't find it in the changelog

**Comment by cruzach** [comment]
Unfortunately, the lottie upgrade requires a bit more on our end than just RN 0.61 support ☹️ (mostly related to Swift)

We're still working on it, and it's looking promising, but no timeline just yet

**Comment by actuallymentor** [comment]
Importing Lottie 3 files causes the previewer to crash without a visible error. The [lottie expo docs](https://docs.expo.io/versions/latest/sdk/lottie/) do not reflect this.

Does it make sense to add a side note there?

**Comment by cruzach** [comment]
Probably, would you like to open a PR for that?

**Comment by actuallymentor** [comment]
Done. #7254

**Comment by ReneMoraales** [comment]
Any progress on the actual issue?

**Comment by cruzach** [comment]
@ReneMoraales unfortunately, no. The move to support Swift is going to take a bit of work, and right now our primary focus is on other things

But, if you do need to use the most recent version of lottie, you should be able to in the bare workflow

**Comment by jciccio** [comment]
Hey same issue here, some Lottie animations makes the Android app crash :(

**Comment by prajwolrg** [comment]
Anyone came up with a fix on this issue?

**Comment by prajwolrg** [comment]
Is there any way to convert lottie file to old JSON format? when you have the lottie JSON file downloaded from [lottiefiles.com](lottifiles.com)

**Comment by Cookizza** [comment]
I'm looking for the same if anyone has a way to convert json to the old format to fix this :(

**Comment by kimfucious** [comment]
https://github.com/react-native-community/lottie-react-native/issues/661#issuecomment-675467218

**Comment by jciccio** [comment]
Question: are still any plans to fix this one? Or any workaround to load animations on Android (downloaded from lottiefiles) without making it crash the app?

**Comment by MorenoMdz** [comment]
> Question: are still any plans to fix this one? Or any workaround to load animations on Android (downloaded from lottiefiles) without making it crash the app?

I second this, having proper support for lottie should be way higher in the expo team priority in my opinion.

**Comment by brendonhc** [comment]
> Unfortunately, the lottie upgrade requires a bit more on our end than just RN 0.61 support  (mostly related to Swift)
> 
> We're still working on it, and it's looking promising, but no timeline just yet

Hey there! Now we are on SDK 39 with Support for iOS 14, React Native 0.63, can you help us with lottie v3 upgrade to stop the app crashes?

**Comment by brendonhc** [comment]
> Unfortunately, the lottie upgrade requires a bit more on our end than just RN 0.61 support  (mostly related to Swift)
> 
> We're still working on it, and it's looking promising, but no timeline just yet

Hey there! Now we are on SDK 39 with Support for iOS 14, React Native 0.63, can you help us with lottie v3 upgrade to stop the app crashes? @cruzach @changLiuUNSW @actuallymentor @SimenB 
Thanks in advance!

**Comment by cruzach** [comment]
@brendonhc unfortunately iOS 14 and RN 0.63 don't make supporting Swift any easier 😞  Right now, if you need to use lottie v3 files, your best bet is to eject to the bare workflow (upgrading to Lottie v3 is on the roadmap for next year)

**Comment by huy-lv** [comment]
@cruzach  I know how you feel because I have worked with react native and native lib. But is there anyway to know which lottie file require v3? which one not?

**Comment by cruzach** [comment]
Available now in the SDK 41 beta! https://blog.expo.io/expo-sdk-41-beta-is-now-available-958edcd78991

**Comment by SimenB** [comment]
Yay 🎉
